src/components/Route.jsx

```diff
@@ -13,7 +13,17 @@ class Route extends Component {
 
   _handleFavoriteClick = () => {
     this.setState({
-      favorite: !this.state.favorite
+      favorite: !this.state.favorite 
+    })   
+  }
+
+  componentDidUpdate = () => {
```

Comment:
I put the local storage.setItem inside of the componentDidUpdate to make sure the favorite state was loaded after the click before it was saved to storage because setState takes time and it would save it as false otherwise.